code/modules/mob/living/carbon/carbon.dm

```diff
@@ -274,17 +272,45 @@
 	skill_mod += 0.8 * (get_skill_value(SKILL_HAULING) - SKILL_MIN)/(SKILL_MAX - SKILL_MIN)
 	throw_range *= skill_mod
 
+	var/multiz_throwing = FALSE
+	if((get_z(target) > get_z(src)))
+		var/turf/T = GetAbove(src)
+		if(!istype(T) || !T.is_open() || !T.CanZPass(item, UP))
+			return
+
+		if(!(item.w_class < ITEM_SIZE_GARGANTUAN))
+			to_chat(src, SPAN_WARNING("\The [item] is too huge to throw!"))
+			return
+
+		if(!do_after(src, 2 SECONDS))
+			to_chat(src, SPAN_WARNING("You were interrupted!"))
+			return
+		else
+			multiz_throwing = TRUE
+
+	if(!unEquip(item))
+		return
+
+	if(!item || !isturf(item.loc))
+		return
+
 	//actually throw it!
 	src.visible_message("<span class='warning'>[message]</span>", range = min(itemsize*2,world.view))
 
 	if(!src.lastarea)
 		src.lastarea = get_area(src.loc)
+
 	if(!check_space_footing())
 		if(prob((itemsize * itemsize * 10) * MOB_SIZE_MEDIUM/src.mob_size))
 			var/direction = get_dir(target, src)
 			step(src,direction)
 			space_drift(direction)
 
+	if(multiz_throwing)
+		item.obj_flags |= OBJ_FLAG_NOFALL
```

Comment:
Why are you messing with `OBJ_FLAG_NOFALL`? This is the flag for objs to keep mobs from falling through open space (eg a lattice or catwalk)

Reply:
Because object instantly fall on focemove.

Reply:
The problem is that it's not being used properly here. This prevents _other_ objects (actually movables, comment is wrong) from falling. It preventing _itself_ from falling is accidental at best, and it will also prevent other stuff from falling for the duration.

The real way to solve this is probably to override `/obj/item/can_fall` to return `FALSE` if currently being thrown (I believe there's a var for that).

Reply:
Did something similar to this, there's a throwing var you can set.

Reply:
on neb throwing var is a datum, which you can't just TRUE/FALSE

Reply:
You can check if it's null or not to achieve the same goal.

Reply:
The issue, I believe, is that this won't work because the var isn't set at this point: this is all happening _prior_ to the throw proper (which happens at the very end of the proc).

Which begs the question of why it's not in the throw handler itself.